pkg/utils/packet.go

```diff
@@ -168,7 +168,7 @@ func AnnounceIPs(ifName string, ipConfigs []*current.IPConfig) error {
 
 	// Retrieve the interface name in the container.
 	linkObj, err := myNetLink.LinkByName(ifName)
-	if err != nil {
+	if err != nil || linkObj.Attrs() == nil {
```

Comment:
is there really a case where netlink LinkByName completes successfully but returned Attr() is nil ?

I think not, and if there is, i think we should panic as it would be an un-expected behaviour of netlink (or any other impl of the interface)

IMO this additional condition should be removed.

Reply:
@adrianchiris 
thanks for review.
i read the ```netlink LinkByName``` code again.
```
func (h *Handle) linkByNameDump(name string) (Link, error) {
	links, err := h.LinkList()
	if err != nil {
		return nil, err
	}

	for _, link := range links {
		if link.Attrs().Name == name {
			return link, nil
		}
	}
	return nil, LinkNotFoundError{fmt.Errorf("Link %s not found", name)}
}
```
the lib also not judge the link.Attrs()==nil , So we may relay on it .
and delete the case for nil